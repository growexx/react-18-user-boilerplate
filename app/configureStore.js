import { configureStore } from '@reduxjs/toolkit';
import { createBrowserHistory } from 'history';
import { createReduxHistoryContext } from 'redux-first-history';

import { repoApi } from './containers/HomePage/reposApiSlice';
import { appSlice } from './containers/App/slice';
import { apiSlice } from './apiSlice';
import languageSlice from './containers/LanguageProvider/slice';
import loginSlice from './containers/Auth/Login/slice';
import { authApi } from './containers/Auth/Login/authApiSlice';
import twoFactorAuthSlice from './containers/Auth/TwoFactorAuthentication/slice';

const { createReduxHistory, routerMiddleware, routerReducer } =
  createReduxHistoryContext({ history: createBrowserHistory() });

const store = configureStore({
  reducer: {
    router: routerReducer,
    app: appSlice,
    language: languageSlice,
    login: loginSlice,
    twoFactorAuth: twoFactorAuthSlice,
    repos: repoApi.reducer,
    auth: authApi.reducer,
    [apiSlice.reducerPath]: apiSlice.reducer,
  },
  middleware: (getDefaultMiddleware) => [
    ...getDefaultMiddleware(),
    apiSlice.middleware,
    routerMiddleware,
  ],
  devTools: process.env.NODE_ENV !== 'production',
});

const history = createReduxHistory(store);

export { store, history };
