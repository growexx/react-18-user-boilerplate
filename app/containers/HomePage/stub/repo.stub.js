export const MOCK_REPO_RESPONSE = {
  id: 11116372,
  node_id: 'MDEwOlJlcG9zaXRvcnkxMTExNjM3Mg==',
  name: 'keystone-classic',
  full_name: 'keystonejs/keystone-classic',
  private: false,
  owner: {
    login: 'keystonejs',
    id: 6118534,
    node_id: 'MDEyOk9yZ2FuaXphdGlvbjYxMTg1MzQ=',
    avatar_url: 'dummy_url',
    gravatar_id: '',
    url: 'dummy_url',
    html_url: 'dummy_url',
    followers_url: 'dummy_url',
    following_url: 'dummy_url',
    gists_url: 'dummy_url',
    starred_url: 'dummy_url',
    subscriptions_url: 'dummy_url',
    organizations_url: 'dummy_url',
    repos_url: 'dummy_url',
    events_url: 'dummy_url',
    received_events_url: 'dummy_url',
    type: 'Organization',
    site_admin: false,
  },
  html_url: 'dummy_url',
  description: 'Node.js CMS and web app framework',
  fork: false,
  url: 'dummy_url',
  forks_url: 'dummy_url',
  keys_url: 'dummy_url',
  collaborators_url: 'dummy_url',
  teams_url: 'dummy_url',
  hooks_url: 'dummy_url',
  issue_events_url: 'dummy_url',
  events_url: 'dummy_url',
  assignees_url: 'dummy_url',
  branches_url: 'dummy_url',
  tags_url: 'dummy_url',
  blobs_url: 'dummy_url',
  git_tags_url: 'dummy_url',
  git_refs_url: 'dummy_url',
  trees_url: 'dummy_url',
  statuses_url: 'dummy_url',
  languages_url: 'dummy_url',
  stargazers_url: 'dummy_url',
  contributors_url: 'dummy_url',
  subscribers_url: 'dummy_url',
  subscription_url: 'dummy_url',
  commits_url: 'dummy_url',
  git_commits_url: 'dummy_url',
  comments_url: 'dummy_url',
  issue_comment_url: 'dummy_url',
  contents_url: 'dummy_url',
  compare_url: 'dummy_url',
  merges_url: 'dummy_url',
  archive_url: 'dummy_url',
  downloads_url: 'dummy_url',
  issues_url: 'dummy_url',
  pulls_url: 'dummy_url',
  milestones_url: 'dummy_url',
  notifications_url: 'dummy_url',
  labels_url: 'dummy_url',
  releases_url: 'dummy_url',
  deployments_url: 'dummy_url',
  created_at: '2013-07-02T05:01:55Z',
  updated_at: '2023-10-27T05:49:50Z',
  pushed_at: '2023-07-28T10:11:43Z',
  git_url: 'dummy_url',
  ssh_url: 'dummy_url',
  clone_url: 'dummy_url',
  svn_url: 'dummy_url',
  homepage: 'https://v4.keystonejs.com',
  size: 151918,
  stargazers_count: 14667,
  watchers_count: 14667,
  language: 'JavaScript',
  has_issues: true,
  has_projects: true,
  has_downloads: true,
  has_wiki: true,
  has_pages: true,
  has_discussions: false,
  forks_count: 2287,
  mirror_url: null,
  archived: false,
  disabled: false,
  open_issues_count: 359,
  license: {
    key: 'mit',
    name: 'MIT License',
    spdx_id: 'MIT',
    url: 'dummy_url',
    node_id: 'MDc6TGljZW5zZTEz',
  },
  allow_forking: true,
  is_template: false,
  web_commit_signoff_required: false,
  topics: ['cms', 'javascript', 'keystonejs', 'nodejs'],
  visibility: 'public',
  forks: 2287,
  open_issues: 359,
  watchers: 14667,
  default_branch: 'master',
};
