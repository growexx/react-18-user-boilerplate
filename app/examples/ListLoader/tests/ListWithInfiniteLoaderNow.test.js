/* eslint-disable react/prop-types */
import React from 'react';
import { render, waitFor } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { HistoryRouter as Router } from 'redux-first-history/rr6';
import history from 'utils/history';
import request from 'utils/request';
import { store } from 'configureStore';
import ListWithInfiniteLoader from '../ListWithInfiniteLoader';
import { responseWithLargeDataSet } from '../stub/test.stub';

let globalStore;
jest.mock('utils/request');
jest.mock('react-infinite-scroller', () => {
  function LoadMore({ children, loadMore }) {
    return (
      <div data-testid="infinite-scroll">
        {children}
        <button onClick={loadMore} data-testid="load-more-button" type="submit">
          Load More
        </button>
      </div>
    );
  }
  return LoadMore;
});

const componentWrapper = () =>
  render(
    <Provider store={globalStore}>
      <IntlProvider locale="en">
        <Router history={history}>
          <ListWithInfiniteLoader />
        </Router>
      </IntlProvider>
    </Provider>,
  );

describe('<ListWithInfiniteLoader />', () => {
  beforeAll(() => {
    globalStore = store;
  });

  it('should render and match the snapshot', () => {
    request.mockImplementation(() => Promise.resolve({ status: 1 }));
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
  it('should render and match the snapshot', async () => {
    request.mockImplementation(() =>
      Promise.resolve({
        status: 1,
        results: [
          {
            name: {
              last: 'testInfiniteLoader',
              email: 'test@234.com',
            },
          },
        ],
      }),
    );
    const {
      container: { firstChild },
    } = componentWrapper();
    await waitFor(() => expect(request).toHaveBeenCalledTimes(2));
    expect(firstChild).toMatchSnapshot();
  });

  it('should load more items when scrolling', async () => {
    request.mockImplementation(() => Promise.resolve(responseWithLargeDataSet));
    const { getByTestId } = componentWrapper();
    waitFor(() => {
      getByTestId('load-more-button').click();
    });
    await waitFor(() => expect(request).toHaveBeenCalledTimes(4));
  });
});
