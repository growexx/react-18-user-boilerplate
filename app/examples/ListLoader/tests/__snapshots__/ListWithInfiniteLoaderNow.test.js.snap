// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListWithInfiniteLoader /> should render and match the snapshot 1`] = `
.c0 .demo-infinite-container {
  overflow: auto;
  border-radius: 4px;
}

.c0 .demo-loading-container {
  position: absolute;
  bottom: 40px;
  width: 100%;
  text-align: center;
}

<div
  class="c0"
>
  <div
    class="demo-infinite-container"
  >
    <div>
      <div
        class="ant-list ant-list-split"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <ul
              class="ant-list-items"
            >
              <li
                class="ant-list-item"
              >
                <div
                  class="ant-list-item-meta"
                >
                  <div
                    class="ant-list-item-meta-avatar"
                  >
                    <span
                      class="ant-avatar ant-avatar-circle ant-avatar-image"
                    >
                      <img
                        src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
                      />
                    </span>
                  </div>
                  <div
                    class="ant-list-item-meta-content"
                  >
                    <h4
                      class="ant-list-item-meta-title"
                    >
                      testInfiniteLoader
                    </h4>
                    <div
                      class="ant-list-item-meta-description"
                    >
                      test@234.com
                    </div>
                  </div>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
